src/Microsoft.SymbolStore/KeyGenerators/KeyGenerator.cs

```diff
@@ -99,7 +99,7 @@ public virtual bool IsDump()
         /// <returns>key</returns>
         protected static SymbolStoreKey BuildKey(string path, string id, bool clrSpecialFile = false, IEnumerable<PdbChecksum> pdbChecksums = null)
         {
-            string file = Uri.EscapeDataString(GetFileName(path).ToLowerInvariant());
+            string file = GetFileName(path).ToLowerInvariant();
```

Comment:
Doing this changes the key for file names. Any file that had these characters will have the key changed on windows: `#[]@!$&'()+,;=`. The parenthesis is the only one that I've seen used in real life. They used to be escaped and now the URL won't escape them. Some of which are invalid URL encoding at the HTTP Store part. While this is the correct part for the index, it might cause issues with the URI generated (say a file named `build+1.2.3.dll` for versioned files like our nupkgs do).

Reply:
The uploader treats the index opaquely and it shouldn't be escaped. The symbol server was incorrectly indexing the file with the escaped index and the downloaders (windbg or dotnet-symbol) couldn't find it.  

I don't think it is a problem with the difference between EscapeDataString and EscapeUriString (:/?) when the index does get sent by the downloaders using symstore.

Reply:
> The uploader treats the index opaquely and it shouldn't be escaped. The symbol server was incorrectly indexing the file with the escaped index.

This is indeed what should happen for the key. We should never touch the file name for the key-building. What I am raising up here is that the behavior has changed, and that we could now build malformed URIs (especially on Unix). That being said, I'm OK with this as is.

Reply:
I think I did find a problem that you are describing testing on this windows dump with the module "xd (dev).exe" in it.  The new code created the index: `xd (dev).exe/5d5c549511d9b000/xd (dev).exe` but the () didn't get escaped when the URI request was sent `http://msdl.microsoft.com/download/symbols/xd%20(dev).exe/5d5c549511d9b000/xd%20(dev).exe`

There is also a SymbolStoreKey.IsKeyValid function that the cache/directory symbol stores use that needs to be changed to allow more special characters that were escaped before.

Reply:
This would be fixed by using EscapeDataString except for the part that is escapes "/".

Reply:
Maybe escaping "/" is ok because it will be unescaped back to / on the symbol server before it looks the index up?

Reply:
I don't know how symbol servers do request routing. However, my guess is no. And as you said, I constructed this test: `foo&bar (1.3.preview+sha256)#hello.dll`. Generated an invalid URI before and after. Looks like a space before a parenthesis is doing weird things in URI encoding...

In you case, I see `xd (dev)` generating `"xd%20%28dev%29.exe/AAAAAAAAAAAAAA/xd%20%28dev%29.exe"`, but then the URI encoding does `{http://msdl.microsoft.com/download/symbols/xd %28dev%29.exe/AAAAAAAAAAAAAA/xd %28dev%29.exe}`